src/liberty/devCommands.ts

```diff
@@ -102,6 +102,121 @@ export async function startDevMode(libProject?: LibertyProject | undefined): Pro
     }
 }
 
+
+export async function removeProject(uri: vscode.Uri): Promise<void> {
+    const projectProvider: ProjectProvider = ProjectProvider.getInstance();
+    if (undefined !== uri && undefined !== uri.fsPath ) {
+        // Right mouse clicked on a root folder, or on empty space with only one folder in workspace.
+        // Add project if:
+        // 1. Project has build files (pom.xml or build.gradle)
+        // 2. Not in liberty dashboard
+        // Once added, presist the data in workspace storage.\
+        // check if the path is in current list
+        const file = projectProvider.isPathExistsInPersistedProjects(uri.fsPath);
+        if (undefined !== file) {
+            const yes = localize("confirmation.button.label.yes");
+            const no = localize("confirmation.button.label.no");
+            vscode.window
+                .showInformationMessage(localize("remove.custom.project.confirmation", uri.fsPath), yes, no)
+                .then(answer => {
+                    if (answer === yes) {
+                    // delete and save
+                        projectProvider.removeInPersistedProjects(file);
+                        vscode.window
+                            .showInformationMessage(localize("remove.custom.project.successful"));
+                            projectProvider.fireChangeEvent();
+                    }
+                });
+        } else {
+            const message = localize("remove.custom.project.not.in.list");
+            console.error(message);
+            vscode.window.showInformationMessage(message);
+        }
+    } else {
+        // clicked on the empty space and workspace has more than one folders, or
+        // from command palette
+        // Display the list of current cusomer added project for user to select.
+        const items: LibertyProjectQuickPickItem[] = [];
+        projectProvider.getUserAddedProjects().forEach(function (item) {
+            const qpItem = new LibertyProjectQuickPickItem(item.label,
+                item.path);
+            items.push(qpItem);
+        });
+        if ( items.length === 0 ) {
+            const message = localize("remove.custom.project.empty.list");
```

Comment:
Same as above, users should be able to remove automatically added projects from the dashboard too.

Reply:
see above comment